charms/kfp-profile-controller/tests/integration/test_charm.py

```diff
@@ -31,10 +36,19 @@ async def test_build_and_deploy(ops_test: OpsTest):
     image_path = METADATA["resources"]["oci-image"]["upstream-source"]
     resources = {"oci-image": image_path}
 
+    # Deploy the admission webhook to apply the PodDefault CRD required by the charm workload
+    await ops_test.model.deploy(entity_url="admission-webhook", channel="latest/edge", trust=True)
+    # TODO: The webhook charm must be active before the metacontroller is deployed, due to the bug
+    # described here: https://github.com/canonical/metacontroller-operator/issues/86
+    # Drop this wait_for_idle once the above issue is closed
+    await ops_test.model.wait_for_idle(apps=["admission-webhook"], status="active")
+
     await ops_test.model.deploy(
         entity_url="metacontroller-operator",
         # TODO: Revert once metacontroller stable supports k8s 1.22
         channel="latest/edge",
+        # TODO: Drop once https://github.com/canonical/metacontroller-operator/issues/84 is closed
+        config={"metacontroller-image": "docker.io/metacontrollerio/metacontroller:v2.0.4"},
```

Comment:
Lets first merge the metacontroller so we dont need to do this

Reply:
Absolutely!